lib/srv/db/cloud/meta.go

```diff
@@ -99,26 +99,28 @@ func (m *Metadata) fetchRDSMetadata(ctx context.Context, database types.Database
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	// First try to fetch the RDS instance metadata.
-	metadata, err := fetchRDSInstanceMetadata(ctx, rds, database.GetAWS().RDS.InstanceID)
-	if err != nil && !trace.IsNotFound(err) && !trace.IsAccessDenied(err) {
```

Comment:
I don't see NotFound and AccessDenied errors being gracefully handled here anymore. Are they handled on the caller side?

Reply:
In the old logic, when error is `IsNotFound` or `IsAccessDenied`, it will go into the next if statement `metadata == nil` and eventually does a `fetchRDSClusterMetadata`.

In the new logic, `fetchRDSClusterMetadata` will be called
- when `ClusterID` is parsed properly and set, `fetchRDSClusterMetadata` will be called before this instance fetch
- when instance is FOUND, and it belongs to cluster, `fetchRDSClusterMetadata` will be called after this instance fetch

all fetch errors including `IsNotFound` and `IsAccessDenied` propagates to `Metadata.Update` and we log a warning
https://github.com/gravitational/teleport/blob/a8a57b19f813960b666b2f35c859d3557eb9d68a/lib/srv/db/server.go#L299-L300